### PR TITLE
textbox additions

### PR DESCRIPTION
**Pull request by Maferep**
- predictive text wrapping
- in-text pausing ({p600})
- skipping in-text pauses
- textbox printing changes

### CODE REVIEW

File: System/UI/Textbox.gd [hunk 1]
```diff
@@ -91,19 +208,15 @@ func get_next_pack():
...
 		if not found_bracket and i > 0 and c == '{':
-			text_to_print = text_to_print.substr(i)
-			return Pack.new(TEXT_PACK, pack.left(pack.length()-1))
+			return [TextPack.new(pack.left(pack.length()-1), self),text_to_print.substr(i)]
 		i += 1
-	text_to_print = ""
-	return Pack.new(TEXT_PACK, pack)
-
-# TODO finish execute markup base commands
-# TODO execute macros
```
Comment (saluk) [inline System/UI/Textbox.gd:104]: Try not to erase TODO comments that still haven't been implemented!

File: System/UI/Textbox.gd [hunk 2]
```diff
@@ -116,82 +229,61 @@ func execute_markup(pack:Pack):
...
+		text_to_print = v[1]
+	packs.append(next_pack)
+	return packs
+
+func _set_speaking_animation(name):
+	for character in Commands.get_speaking_char():
+		character.play_state("talk")
+		
+func strip_bbcode(source:String) -> String:
+	var regex = RegEx.new()
```
Comment (saluk) [inline System/UI/Textbox.gd:256]: I'm not a fan of having to do this, but I think it's probably cleaner than bookeeping the length elsewhere so I'm OK with it. On the plus side it makes it easier if we want to support arbitrary bbcode in the future.

**Review by saluk — COMMENTED** [review]
This is great! There's a few things I don't like about the approach but I think it's good enough for now. My reservations are mainly future proofing: I think this might make it more challenging to add commands and impossible to have more dynamic text in the future, in those cases where that might be desired (though it probably isn't necessary).

A perhaps more hybrid approach would be to get the next pack, generate its text (whether that be a word or many words), and then add ALL of the text to the Label, still using the approach of using visible_characters to control the actual character by character print speed.

Let me know what you think, I'm OK to iterate on this some more or to merge it and we can always swing back to refactor if necessary.

**Comment by Maferep** [comment]
> This is great! There's a few things I don't like about the approach but I think it's good enough for now. My reservations are mainly future proofing: I think this might make it more challenging to add commands and impossible to have more dynamic text in the future, in those cases where that might be desired (though it probably isn't necessary).
> 
> A perhaps more hybrid approach would be to get the next pack, generate its text (whether that be a word or many words), and then add ALL of the text to the Label, still using the approach of using visible_characters to control the actual character by character print speed.
> 
> Let me know what you think, I'm OK to iterate on this some more or to merge it and we can always swing back to refactor if necessary.

I see! I think I'll sit down and address some of your comments. I'll give it a week or two to see if i can find a future proofing solution then get back to you :)

**Pull request closed without merging**